view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
     return sendPinotTableCreationRequest(
         pinotTableSpec.getControllerHost(),
         pinotTableSpec.getControllerPort(),
-        tableConfig.toJsonString());
+        tableConfig.toJsonString(),
+        tableConfig.getTableName());
   }
 
   public static boolean sendPinotTableCreationRequest(
-      String controllerHost, String controllerPort, final String tableConfig) {
+      String controllerHost,
+      String controllerPort,
+      final String tableConfig,
+      final String tableName) {
+    String controllerAddress = getControllerAddressForTableCreate(controllerHost, controllerPort);
+    LOGGER.info("Trying to send table creation request {} to {}. ", tableConfig, controllerAddress);
+    int responseCode = sendRequest("POST", controllerAddress, tableConfig);
+    if (responseCode == 409) {
```

Comment:
static var for response code 409

Reply:
Some comment around what this means and how we proceed

Reply:
Okay